src/image/generators/starVsTheForcesOf.ts

```diff
@@ -1,51 +1,65 @@
 import { BaseImageGenerator } from '@blargbot/image/BaseImageGenerator';
 import { ImageWorker } from '@blargbot/image/ImageWorker';
 import { ImageResult, StarVsTheForcesOfOptions } from '@blargbot/image/types';
-import { ColorAction } from '@jimp/plugin-color';
-import colorThief from 'color-thief-jimp';
-import Jimp from 'jimp';
+import sharp from 'sharp';
 
 export class StarVsTheForcesOfGenerator extends BaseImageGenerator<'starVsTheForcesOf'> {
     public constructor(worker: ImageWorker) {
         super('starVsTheForcesOf', worker);
     }
 
     public async execute({ avatar }: StarVsTheForcesOfOptions): Promise<ImageResult> {
-        const avatarImg = await this.getRemoteJimp(avatar);
-        avatarImg.resize(700, 700);
-        const color = colorThief.getColor(avatarImg);
-        //color = color.map(a => a / 2);
-        const lowest = Math.min(color[0], color[1], color[2]);
-        const mappedColor = color.map(a => Math.min(a - lowest, 32));
-        this.worker.logger.debug(mappedColor);
-        const bgImg = await this.generateJimp(avatarImg, x => {
+        const avatarImg = sharp(await this.getRemote(avatar)).resize(700, 700);
+        const bgImg = sharp(await this.generate(await avatarImg.toBuffer(), x => {
             x.out('-matte').out('-virtual-pixel').out('transparent');
-            x.out('-extent');
-            x.out('1468x1656');
-            x.out('-distort');
-            x.out('Perspective');
-            x.out('0,0,0,208  700,0,1468,0  0,700,0,1326  700,700,1468,1656');
-        });
-        bgImg.resize(734, 828);
+            x.out('-extent').out('1468x1656');
+            x.out('-distort').out('Perspective').out('0,0,0,208  700,0,1468,0  0,700,0,1326  700,700,1468,1656');
+        })).resize(734, 828)
+            .extract({ left: 0, top: 104, width: 600, height: 540 });
 
-        const foreground = await this.getLocalJimp('starvstheforcesof.png');
-        foreground.resize(960, 540);
-        const actions: ColorAction[] = [];
-        if (mappedColor[0] > 0)
-            actions.push({ apply: 'red', params: [mappedColor[0]] });
-        if (mappedColor[1] > 0)
-            actions.push({ apply: 'green', params: [mappedColor[1]] });
-        if (mappedColor[2] > 0)
-            actions.push({ apply: 'blue', params: [mappedColor[2]] });
-        foreground.color(actions);
-        const img = new Jimp(960, 540);
-        bgImg.crop(0, 104, 600, 540);
-        img.composite(bgImg, 430, 0);
-        img.composite(foreground, 0, 0);
+        const foreground = sharp(this.getLocalResourcePath('starvstheforcesof.png'))
+            .resize(960, 540)
+            .linear(1, 0); // TODO: need latest version of sharp
```

Comment:
what's that comment about?

Reply:
In the jimp version, the foreground gets a tint based on the dominant color of the image in the background. The current version of sharps linear method doesnt support transforming the channel values by different amounts. The latest version has the option to do this, but it was only released 2 days ago and the @types/sharp package hasnt been updated yet.

Reply:
Would it be a good idea to wait for the update of @types/sharp ?

Reply:
Im not sure when that will be, I dont see a PR for it yet. This is a minor loss of functionality, I think its fine to go ahead with this and update later once the package is released